Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -276,159 +282,66 @@ public async Task<IEnumerable<SubmissionForProfileServiceModel>> GetForProfileBy
+                problem!.ProblemGroup.ContestId,
                 currentUser.Id!,
                 model.Official);
-        if (participant == null)
-        {
-            throw new BusinessServiceException(Resources.ContestsGeneral.UserIsNotRegisteredForExam);
-        }
-
-        await this.contestsBusinessService.ValidateContest(
-            participant.Contest,
-            currentUser.Id!,
-            currentUser.IsAdmin,
-            model.Official);
-
-        this.problemsBusinessService.ValidateProblemForParticipant(
-            participant,
-            participant.Contest,
-            model.ProblemId,
-            model.Official);
-
-        // if (official &&
-        //     !this.contestsBusinessService.IsContestIpValidByContestAndIp(problem.ProblemGroup.ContestId, this.Request.UserHostAddress))
-        // {
-        //     return this.RedirectToAction("NewContestIp", new { id = problem.ProblemGroup.ContestId });
-        // }
-
-        this.submissionTypesBusinessService.ValidateSubmissionType(model.SubmissionTypeId, problem);
-
-        if (this.submissionsData.GetUserSubmissionTimeLimit(
-                participant.Id,
-                participant.Contest.LimitBetweenSubmissions) != 0)
-        {
-            throw new BusinessServiceException(Resources.ContestsGeneral.SubmissionWasSentTooSoon);
-        }
-
-        if (problem.SourceCodeSizeLimit < model.Content.Length)
-        {
-            throw new BusinessServiceException(Resources.ContestsGeneral.SubmissionTooLong);
-        }
-
-        if (this.submissionsData.HasUserNotProcessedSubmissionForProblem(problem.Id, currentUser.Id!))
+        var contestValidationResult = this.contestValidationService.GetValidationResult(
+            (participant?.Contest,
+            currentUser.Id, currentUser.IsAdminOrLecturer, model.Official) !);
+        var userSubmissionTimeLimit = this.submissionsData.GetUserSubmissionTimeLimit(
+            participant!.Id,
+            participant.Contest.LimitBetweenSubmissions);
+        var hasUserNotProcessedSubmissionForProblem =
+            this.submissionsData.HasUserNotProcessedSubmissionForProblem(problem.Id, currentUser.Id!);
+
+        var submitSubmissionValidationServiceResult = this.submitSubmissionValidationService.GetValidationResult(
+        (problem,
+        currentUser,
+        participant,
+        contestValidationResult,
+        userSubmissionTimeLimit,
+        hasUserNotProcessedSubmissionForProblem,
+        model));
+
+        if (!submitSubmissionValidationServiceResult.IsValid)
         {
-            throw new BusinessServiceException(Resources.ContestsGeneral
-                .UserHasNotProcessedSubmissionForProblem);
+            throw new BusinessServiceException(submitSubmissionValidationServiceResult.Message);
         }
 
-        var contest = participant.Contest;
-
-        var newSubmission = new Submission
+        Submission newSubmission;
+        if (model.StringContent == null)
```

Comment:
Use automapper and call newSubmission = model.Map<SubmitSubmissionServiceModel>. Then assign values that don't exist in SubmitSubmissionServiceModel like participant. This way you can generally remove the if-else statement. You are not assigning values to IsPublic and IpAddress.

Reply:
Done.